doc/driver.mld

```diff
@@ -1,1124 +1,415 @@
+the [odoc] command with the right arguments throughout the different
+phases. Several drivers for [odoc] exist, such as:
+{{:https://dune.readthedocs.io/en/stable/documentation.html}dune} and
+{{:https://erratique.ch/software/odig}odig}.
 
-The file [deps.mld], which corresponds with the subdirectory [odoc/deps/], will be compiled as follows:
+The [odoc] tools also contains a "reference driver", that is kept up-to-date
+with the latest development of [odoc].
+
+This document explains how to drive [odoc], as of version 3. It is not needed to
+know any of this to {e use} [odoc], it is targeted at driver authors, tools that
+interact with [odoc], or any curious passerby. This includes several subjects:
 
-{@sh skip[
-odoc compile deps.mld -I . --parent page-odoc --child page-stdlib --child page-astring ...
-]}
+- A big picture view of the doc generation model,
+- A unified explanation for the various command line flags,
+- An explanation of the [odoc] pipeline,
+- A convention for building docs for [opam]-installed packages.
 
-The file [odoc_model.mld] will have a child module [Odoc_model]. It will be compiled as follows:
+In addition to the documentation, the reference driver is a good tool to
+understand how to build [odoc] projects. It can be useful to look at the
+implementation code, but it can also help to simply look at all invocation of
+[odoc] during a run of the driver.
 
-{@sh skip[
-odoc compile odoc_model.mld -I . --parent page-odoc --child module-Odoc_model
-]}
+{1:units Units of documentation}
 
-The last type of page contains a list of paths to the source files that should be rendered as HTML. The output will be found as a tree underneath this page. This will be compiled in the following way:
+In its third major version, [odoc] has been improved so that a single
+documentation can work on multiple scenarios, from local switches to big
+monorepos, or the ocaml.org hub of documentation for all packages.
 
-{@sh skip[
-odoc source-tree source.map -I . --parent page-odoc
-]}
+The idea is that we have named units of documentation. We have two kind of them:
+page units, and modules units. Inside the units, everything is managed by
+[odoc]. Outside of the unit, the driver is free to arranged them however they
+like. (Note that however, for building [opam] packages there is a convention to
+follow for the driver.) In order to reference another unit, a documentation
+author can use the name of the unit.
 
-The first few lines of [source.map] will be:
+Different situations will give different meanings to the units. In the case of
+[opam] packages, though, there is a natural meaning to give to those units
+(you'll find more details in the convention for opam-installed packages). Any
+opam package will have an associated "documentation unit", with the name of the
+package. Any of its libraries will have an associated "module unit", with the
+name of the library. Another package can refer to the doc using the package
```

Comment:
I wasn't quite clear here what the module unit does? Is it the page for the module that is the library itself? Is a submodule not also a module unit? If it isn't, might we call module units library units instead (since only whole libraries have them)?

Reply:
Thanks for the review!

I named it "module unit" since it contains modules, just as a "page unit" contains page.

However, the "unit" name is very confusing I think, and was a bad choice, since we already have "OCaml compilation units" for `.cm{t;ti;i}` and sometimes "odoc units" for `.odoc`.

I'll update that to hopefully something less confusing!